src/test/java/org/entando/kubernetes/service/EntandoBundleUtilitiesSecondaryTenantTest.java

```diff
@@ -0,0 +1,48 @@
+package org.entando.kubernetes.service;
+
+import static org.entando.kubernetes.stubhelper.PluginStubHelper.SECONDARY_TENANT_CODE;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import io.fabric8.kubernetes.api.model.EnvVar;
+import io.fabric8.kubernetes.api.model.EnvVarBuilder;
+import io.fabric8.kubernetes.api.model.EnvVarSource;
+import io.fabric8.kubernetes.api.model.EnvVarSourceBuilder;
+import java.util.Collections;
+import java.util.List;
+import org.entando.kubernetes.exception.EntandoValidationException;
+import org.entando.kubernetes.model.bundle.descriptor.plugin.EnvironmentVariable;
+import org.entando.kubernetes.service.digitalexchange.BundleUtilities;
+import org.entando.kubernetes.stubhelper.PluginStubHelper;
+import org.entando.kubernetes.utils.TenantSecondaryContextJunitExt;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+@Tag("unit")
+@ExtendWith(TenantSecondaryContextJunitExt.class)
+class EntandoBundleUtilitiesSecondaryTenantTest {
+
+    @Test
+    void secretOwnedByPrimaryResultsInExceptionBeingThrownIfUsedInOtherTenant() {
+        final String ENVIRONMENT_VARIABLE = "ENVIRONMENT_VARIABLE";
+
+        EnvVarSource envVarSource = new EnvVarSourceBuilder()
+                .withNewSecretKeyRef("key", "pn-hasldk12-8dsjahj2-mypluginname-conf", null)
```

Comment:
please remove the `-conf` suffix since it should be used only in plugin configuration secrets while this one should be a simple referenced secret. Just to avoid confusion while looking at the test in the future